### ISSUE TITLE
Confidence of given pose location

### ISSUE THREAD

**Issue by aarun2211** [introlab/rtabmap#433]
Hello, 
Maybe I am not looking in the right place, but is there a way to query the confidence value of a given robot position measurement in the pose graph? 

Thank you for your answer!

**Comment by matlabbe** [comment]
No, we cannot get this info easily. After an optimization, we can get the marginals of the graph here for example in 3D if we are using g2o:
https://github.com/introlab/rtabmap/blob/2a1b73747078a16b5ba0d6831f7c215ff88bc212/corelib/src/optimizer/OptimizerG2O.cpp#L1122-L1128
In this code, we only get the covariance of the latest node, though it may be possible to iterate over all nodes in the graph. If you want to analyse the graph offline with g2o tools, you could export the graph in *.g2o format.

For GTSAM, marginals are computed here:
https://github.com/introlab/rtabmap/blob/2a1b73747078a16b5ba0d6831f7c215ff88bc212/corelib/src/optimizer/OptimizerGTSAM.cpp#L658-L664
